Aptoide/aptoide-client-v8
# [DT-353] Create home categories list

**What does this PR do?**

   - Creates the categories feature module capable of fetching categories through a use case, repository and network service.
   - Prepares to receive widgets of type "STORE_GROUPS" and bundles of type "CATEGORIES".

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] CategoriesViewModel.kt (inside feature_categories module)

**How should this be manually tested?**

  - Check if the categories list is being fetched correctly. The widget can be received by using the Charles Proxy rewrite set in here: [https://aptoide.atlassian.net/browse/DT-353?focusedCommentId=1212610](url) 

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-353](https://aptoide.atlassian.net/browse/DT-353)

**Code Review Checklist**


[DT-353]: https://aptoide.atlassian.net/browse/DT-353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
